tide_webform.module

```diff
@@ -438,3 +439,188 @@ function tide_webform_webform_element_default_properties_alter(array &$propertie
 function tide_webform_webform_update(EntityInterface $entity) {
   \Drupal::cache()->invalidate('webform_text_fields_default_maxlength');
 }
+
+/**
+ * Implements hook_webform_third_party_settings_form_alter().
+ */
+function tide_webform_webform_third_party_settings_form_alter(array &$form, FormStateInterface $form_state) {
+  $webform = $form_state->getFormObject()->getEntity();
+  $third_party_settings = $webform->getThirdPartySettings('tide_webform_captcha');
+  $user_input = $form_state->getUserInput();
+  $captcha_type = NULL;
+  $allowed_values = [];
+  try {
+    $field_definition = \Drupal::entityTypeManager()->getStorage('field_storage_config')->load('taxonomy_term.field_captcha_type');
+    if ($field_definition) {
+      $allowed_values = $field_definition->getSetting('allowed_values');
+    }
+    else {
+      throw new \Exception('Field storage configuration not found for taxonomy_term.field_captcha_type');
+    }
+  }
+  catch (\Exception $e) {
+    \Drupal::messenger()->addWarning('Unable to load CAPTCHA type options. Please contact the site administrator.');
+    \Drupal::logger('tide_webform')->error('Error loading CAPTCHA type options: @message', ['@message' => $e->getMessage()]);
```

Comment:
Should this logger include the webform id as well for better debugging when looking into the logs. May be something like this - 
`\Drupal::logger('tide_webform')->error('Error loading CAPTCHA type options for webform @webform: @message', ['@webform' => $webform->id(), '@message' => $e->getMessage()]);
`

Reply:
changed. thanks.